wgpu-hal/src/dx12/device.rs

```diff
@@ -312,34 +321,39 @@ impl super::Device {
             size,
             mip_level_count,
             sample_count,
+            allocation: None,
         }
     }
 }
 
 impl crate::Device<super::Api> for super::Device {
-    unsafe fn exit(self, queue: super::Queue) {
+    unsafe fn exit(#[allow(unused_mut)] mut self, queue: super::Queue) {
         self.rtv_pool.lock().free_handle(self.null_rtv_handle);
         unsafe { self.rtv_pool.into_inner().destroy() };
         unsafe { self.dsv_pool.into_inner().destroy() };
         unsafe { self.srv_uav_pool.into_inner().destroy() };
         unsafe { self.sampler_pool.into_inner().destroy() };
         unsafe { self.shared.destroy() };
         unsafe { self.idler.destroy() };
+        #[cfg(feature = "windows_rs")]
```

Comment:
This shouldn't need to be conditional?

Reply:
Clippy complains if it isn't

Reply:
fwiw, you could just write `self.mem_allocator = None`

Reply:
I just copied their example there, which says `drop(allocator); // Explicitly drop before destruction of device.`.

Does setting it to None ensure that it'll be dropped before the device is destroyed, or does it only drop at the end of the block where it was set to `None`?

Reply:
It drops immediately.

Reply:
I personally prefer `.take()` over `= None`, for these kind of things, fwiw. It reads a bit more natural IMO.